client/components/addproduct.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { addProduct } from '../store';
+
+class AddProduct extends Component {
+  constructor() {
+    super();
+    this.state = {};
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+  handleChange(event) {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const title = event.target.title.value;
+    const description = event.target.description.value;
+    const price = event.target.price.value;
+    const quantity = event.target.quantity.value;
+    const weight = event.target.weight.value;
+    const brand = event.target.brand.value;
+
+    this.props.addProduct({
+      title,
+      description,
+      price,
+      quantity,
+      weight,
+      brand,
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <h3>Add A New Product</h3>
+        <form onSubmit={this.handleSubmit}>
+          <div>
+            <input
+              placeholder="Product Title"
+              id="title"
```

Comment:
should we add value field as well, eg. value={this.state.title} ?

Reply:
Added!